### ISSUE TITLE
Support BPF_MAP_TYPE_HASH_OF_MAPS and BPF_MAP_TYPE_ARRAY_OF_MAPS

### ISSUE THREAD

**Issue by dthaler** [microsoft/ebpf-for-windows#375]
This was tracked as part of issue #343.  Splitting into its own issue so it can have its own milestone and assignee.

**Comment by dthaler** [comment]
Support for BPF_MAP_TYPE_HASH_OF_MAPS was blocked on vbpf/ebpf-verifier#228 but that issue is now fixed as verifier support was just merged, so this issue is now unblocked.

**Comment by dthaler** [comment]
Some additional docs are in https://github.com/iovisor/gobpf/issues/120

**Comment by dthaler** [comment]
Still remaining in this issue are:

- [ ] ensure that all inner maps match the one specified by inner_map_idx, much like prog_types have to match in a BPF_MAP_TYPE_PROG_ARRAY.
- [ ] ensure that putting a prog_array in an array of maps adheres to the prog_array contract that any associated progs have to match the type of the calling program.

And one more piece tracked as part of #396: 
- read a map id not fd when UM reads the value (will be done together with issue PROG_ARRAY should contain prog_id when read #396 since also affects prog_arrays).
